lib/pyprojectOverrides.nix

```diff
@@ -0,0 +1,181 @@
+# Copyright lowRISC Contributors.
+# Licensed under the MIT License, see LICENSE for details.
+# SPDX-License-Identifier: MIT
+{pkgs, ...}: let
+  # uv.lock doesn't contain build dependencies, so we need to specify how each
+  # package needs to be built.
+  # https://pyproject-nix.github.io/uv2nix/patterns/overriding-build-systems.html
+  #
+  # This list here is long so that it supports building packages from source if possible.
+  buildSystemOverrides = {
```

Comment:
We have to maintain this list now? That seems kinda unfortunate...

Reply:
This list can be much shorter if we prefer to use wheels where possible, but I still prefer to use source dist if possible..

Poetry I think contains build system info in its lock file, so just need to add missing ones, but uv doesn't contain this info at all.